packages/wordcount/src/index.js

```diff
@@ -0,0 +1,112 @@
+import { extend, flow  } from 'lodash';
+import { defaultSettings } from './defaultSettings'
+import stripTags from './stripTags';
+import transposeAstralsToCountableChar from './transposeAstralsToCountableChar';
+import stripHTMLEntities from './stripHTMLEntities';
+import stripConnectors from './stripConnectors';
+import stripRemovables from './stripRemovables';
+import stripHTMLComments from './stripHTMLComments';
+import stripShortcodes from './stripShortcodes';
+import stripSpaces from './stripSpaces';
+import transposeHTMLEntitiesToCountableChars from './transposeHTMLEntitiesToCountableChars';
+
+
+
+/**
+ * Private function to manage the settings.
+ *
+ * @param type
+ * @param userSettings
+ * @returns {void|Object|*}
+ */
+function loadSettings( type, userSettings ) {
+	let settings = extend( defaultSettings, userSettings  );
+
+	settings.shortcodes = settings.l10n.shortcodes || {};
+
+	if ( settings.shortcodes && settings.shortcodes.length ) {
+		settings.shortcodesRegExp = new RegExp( '\\[\\/?(?:' + settings.shortcodes.join( '|' ) + ')[^\\]]*?\\]', 'g' );
+	}
+
+	settings.type = type || settings.l10n.type;
+
+	if ( settings.type !== 'characters_excluding_spaces' && settings.type !== 'characters_including_spaces' ) {
+		settings.type = 'words';
+	}
+
+	return settings;
+}
+/**
+ * Count some words.
+ *
+ * @param {String} text
+ * @param {String} type
+ * @param {Object}  userSettings
+ *
+ * @returns {Number}
+ */
+export const WordCounter = {
```

Comment:
Should we change the API like proposed before  https://github.com/WordPress/packages/pull/10#issuecomment-313775265

Unless I've missed a comment, it seems we have a consensus to use the functional approach and ensure the Backwards compatibility when importing this in WP (if needed)